amethyst_renderer/src/cam.rs

```diff
@@ -76,6 +79,27 @@ impl Camera {
             std::f32::consts::FRAC_PI_3,
         ))
     }
+
+    /// Transforms position from screen space to camera space
+    pub fn position_from_screen(
+        &self,
+        screen_position: Vector2<f32>,
+        camera_transform: &GlobalTransform,
+        screen_dimensions: &ScreenDimensions,
+    ) -> Vector3<f32> {
+        let screen_x = 2.0 * screen_position.x / screen_dimensions.width() - 1.0;
+        let screen_y = 1.0 - 2.0 * screen_position.y / screen_dimensions.height();
+        let screen_point = Point3::new(screen_x, screen_y, 0.0).to_homogeneous();
+
+        let matrix = camera_transform.0
+            * self
+                .proj
+                .try_inverse()
+                .expect("Camera projection matrix is not invertible")
+            * screen_point;
+
+        Vector3::new(matrix.x, matrix.y, matrix.z)
```

Comment:
I said Vector3 earlier, but I forgot nalgebra has a Point3 type which might make more sense here. Same applies for the input Vector2 with Point2.